template_credentials.go

```diff
@@ -0,0 +1,78 @@
+package transloadit
+
+import (
+	"context"
+)
+
+// Template contains details about a single template.
+type TemplateCredential struct {
+	ID          string                 `json:"id"`
+	Name        string                 `json:"name"`
+	Type        string                 `json:"type"`
+	Content     map[string]interface{} `json:"content"`
+	Created     string                 `json:"created",omitempty`
+	Modified    string                 `json:"modified",omitempty`
+	Deleted     string                 `json:"deleted",omitempty`
+	Stringified string                 `json:"stringified",omitempty`
+}
+
+type TemplateCredentialResponseBody struct {
+	Credential TemplateCredential `json:"credential"`
+	OK         string             `json:"ok"`
+	Message    string             `json:"message"`
+}
+
+// TemplateList contains a list of templates.
+type TemplateCredentialList struct {
+	TemplateCredential []Template `json:"credentials"`
+	OK                 string     `json:"ok"`
+	Message            string     `json:"message"`
+}
+
+// NewTemplate returns a new Template struct with initialized values. This
+// template will not be saved to Transloadit. To do so, please use the
+// Client.CreateTemplate function.
+func NewTemplateCredential() TemplateCredential {
+	return TemplateCredential{
+		Content: make(map[string]interface{}),
+	}
+}
+
+var templateCredentialPrefix = "template_credentials"
+
+// CreateTemplate will save the provided template struct as a new template
+// and return the ID of the new template.
+func (client *Client) CreateTemplateCredential(ctx context.Context, templateCredential TemplateCredential) (string, error) {
+	content := map[string]interface{}{
+		"name":    templateCredential.Name,
+		"type":    templateCredential.Type,
+		"content": templateCredential.Content,
+	}
+	var response TemplateCredentialResponseBody
+	if err := client.request(ctx, "POST", templateCredentialPrefix, content, &response); err != nil {
+		return "", err
+	}
+	templateCredential = response.Credential
```

Comment:
This assignment is not doing much, is it? templateCredential is not a pointer and thus this assignment just manipulates the local variable.